redash/handlers/dashboards.py

```diff
@@ -113,6 +113,43 @@ def post(self):
         return DashboardSerializer(dashboard).serialize()
 
 
+class MyDashboardsResource(BaseResource):
+    @require_permission("list_dashboards")
```

Comment:
```suggestion
    @require_permission("create_dashboard")
```

Reply:
I'm realizing my code is inconsistent on whether to require list_ or create_, so I need to make some sort of change. I went back and forth on this. Here's my thinking:

Currently with only list_dashboards permission, I can list all the dashboards and see who made them. Obviously, I could manually go through them and find all my own dashboards; that's what users do today. MyDashboardsResource just automates this, returning a list already filtered to my own dashboards. So there's no data leakage here in just requiring list_dashboards.

I think the difference only matters in the scenario where someone once had create_dashboards, so was able to make some dashboards, but this was later revoked. At this point, even though they can't make new dashboards, I think we still want them to be able to filter the list to the old dashboards they made. It would probably feel like a bug if they couldn't.

Does that make sense or am I misunderstanding how the permissions work?

Reply:
+1 for this. Also, assume we have a user with `create_dashboard` permission, and then we revoke `create_dashboard ` permission for them. That user should be able to see dashboards they already created (and as @justint-db mentioned) they still can - by walking through the list of all dashboards. So I suggest to allow "My Dashboards" UI and API for users with `list_dashboards`, but need to adjust DashboardListEmptyState component - show 'Create..." button only for users with `create_dashboards` permission. @arikfr WDYT?

Reply:
Updated the PR to match Levko's suggestion.

Reply:
SGTM